.github/workflows/unit_test.yml

```diff
@@ -0,0 +1,28 @@
+name: Unit Testing
+
+on: [push]
```

Comment:
Wouldn't it be more useful to run these on pull requests ? So the main branch is always ready to be released.

Reply:
Yes, I think so. I don't want to make too many assumptions about the code base. :)

- https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows

We can change to something like this. It would probably be best to modify with a code promotion workflow eventually.

```yml
on:
  # Trigger the workflow on push or pull request,
  # but only for the master branch
  push:
    branches:
      - master
  pull_request:
    branches:
      - master
```

Reply:
sorry, current now we don't want add hook at pr step, in the future we will improve that,